rust/abacus-base/src/contract_sync/outbox.rs

```diff
@@ -260,160 +307,100 @@ mod test {
-                    .in_sequence(&mut seq)
-                    .return_once(|| Ok(170));
-                mock_indexer
-                    .expect__fetch_sorted_messages()
-                    .times(1)
-                    .with(eq(121), eq(140))
-                    .in_sequence(&mut seq)
-                    .return_once(move |_, _| Ok(vec![(m3_clone, meta()), (m5_clone, meta())]));
+                expect_current_position!(121);
+                expect_fetches_range!(118, 140, &[&m3, &m5]);
+                expect_backtrack!(121);
 
                 // Indexer retries, the same range in hope of filling the gap,
                 // which it now does successfully
-                mock_indexer
-                    .expect__get_finalized_block_number()
-                    .times(1)
-                    .in_sequence(&mut seq)
-                    .return_once(|| Ok(170));
-                mock_indexer
-                    .expect__fetch_sorted_messages()
-                    .times(1)
-                    .with(eq(121), eq(140))
-                    .in_sequence(&mut seq)
-                    .return_once(move |_, _| Ok(vec![(m3, meta()), (m4, meta()), (m5, meta())]));
+                expect_current_position!(121);
+                expect_fetches_range!(121, 140, &[&m3, &m4, &m5]);
 
                 // Indexer continues with the next block range, which happens to be empty
-                mock_indexer
-                    .expect__get_finalized_block_number()
-                    .times(1)
-                    .in_sequence(&mut seq)
-                    .return_once(|| Ok(180));
-                mock_indexer
-                    .expect__fetch_sorted_messages()
-                    .times(1)
-                    .with(eq(141), eq(160))
-                    .in_sequence(&mut seq)
-                    .return_once(move |_, _| Ok(vec![]));
-
-                // Indexer catches up with the tip
-                mock_indexer
-                    .expect__get_finalized_block_number()
-                    .times(1)
-                    .in_sequence(&mut seq)
-                    .return_once(|| Ok(180));
-                mock_indexer
-                    .expect__fetch_sorted_messages()
-                    .times(1)
-                    .with(eq(161), eq(180))
-                    .in_sequence(&mut seq)
-                    .return_once(move |_, _| Ok(vec![]));
+                expect_current_position!(141);
+                expect_fetches_range!(141, 160, &[]);
 
                 // Stay at the same tip, so no other fetch_sorted_messages calls are made
-                mock_indexer
-                    .expect__get_finalized_block_number()
-                    .returning(|| Ok(180));
+                mock_cursor.expect__current_position().returning(|| 161);
+                mock_cursor.expect__next_range().returning(|| {
+                    Box::pin(async move {
+                        sleep(Duration::from_secs(100)).await;
```

Comment:
why sleep 100 secs? does this mean that tests take 100 seconds longer now

Reply:
No, actually, the tests now take only 20ms. This sleep is because it should be longer than the timeout on the tests since we don't really want to yield it at all. Adding a comment to explain this.